diff --git a/src/compute/block.rs b/src/compute/block.rs
index 4029a268..0d253d23 100644
--- a/src/compute/block.rs
+++ b/src/compute/block.rs
@@ -545,6 +545,7 @@ fn compute_inner(
         #[cfg(feature = "content_size")]
         content_size: Size::ZERO,
         first_baselines: Point::NONE,
+        last_baselines: Point::NONE,
         top_margin: if own_margins_collapse_with_children.start {
             first_child_top_margin_set
         } else {
diff --git a/src/compute/flexbox.rs b/src/compute/flexbox.rs
index d7363e30..a06e8376 100644
--- a/src/compute/flexbox.rs
+++ b/src/compute/flexbox.rs
@@ -422,11 +422,28 @@ fn compute_preliminary(tree: &mut impl LayoutFlexboxContainer, node: NodeId, inp
             })
     };
 
-    LayoutOutput::from_sizes_and_baselines(
+    let last_vertical_baseline = if flex_lines.is_empty() {
+        None
+    } else {
+        let last_line = flex_lines.last().unwrap();
+        last_line
+            .items
+            .iter()
+            .find(|item| constants.is_column || item.align_self == AlignSelf::LAST_BASELINE)
+            .or_else(|| last_line.items.iter().next())
+            .map(|child| {
+                let offset_vertical = if constants.is_row { child.offset_cross } else { child.offset_main };
+                offset_vertical + child.baseline
+            })
+    };
+
+    let mut output = LayoutOutput::from_sizes_and_baselines(
         constants.container_size,
         inflow_content_size.f32_max(absolute_content_size),
         Point { x: None, y: first_vertical_baseline },
-    )
+    );
+    output.last_baselines = Point { x: None, y: last_vertical_baseline };
+    output
 }
 
 /// Compute constants that can be reused during the flexbox algorithm.
@@ -1450,14 +1467,14 @@ fn calculate_children_base_lines(
     for line in flex_lines {
         // If a flex line has one or zero items participating in baseline alignment then baseline alignment is a no-op so we skip
         let line_baseline_child_count =
-            line.items.iter().filter(|child| child.align_self == AlignSelf::BASELINE).count();
+            line.items.iter().filter(|child| child.align_self == AlignSelf::BASELINE || child.align_self == AlignSelf::LAST_BASELINE).count();
         if line_baseline_child_count <= 1 {
             continue;
         }
 
         for child in line.items.iter_mut() {
             // Only calculate baselines for children participating in baseline alignment
-            if child.align_self != AlignSelf::BASELINE {
+            if child.align_self != AlignSelf::BASELINE && child.align_self != AlignSelf::LAST_BASELINE {
                 continue;
             }
 
@@ -1492,7 +1509,11 @@ fn calculate_children_base_lines(
                 Line::FALSE,
             );
 
-            let baseline = measured_size_and_baselines.first_baselines.y;
+            let baseline = if child.align_self == AlignSelf::LAST_BASELINE {
+                measured_size_and_baselines.last_baselines.y
+            } else {
+                measured_size_and_baselines.first_baselines.y
+            };
             let height = measured_size_and_baselines.size.height;
 
             child.baseline = baseline.unwrap_or(height) + child.margin.top;
@@ -1847,6 +1868,18 @@ fn align_flex_items_along_cross_axis(
                 }
             }
         }
+        AlignItemsKeyword::LastBaseline => {
+            if constants.is_row {
+                max_baseline - child.baseline
+            } else {
+                let baseline_column_should_reverse = cross_axis_should_reverse && !constants.is_wrap;
+                if constants.is_wrap_reverse ^ baseline_column_should_reverse {
+                    free_space
+                } else {
+                    0.0
+                }
+            }
+        }
         AlignItemsKeyword::Stretch => {
             if constants.is_wrap_reverse ^ cross_axis_should_reverse {
                 free_space
@@ -2453,11 +2486,11 @@ fn perform_absolute_layout_on_absolute_children(
                 (AlignItemsKeyword::End, true) => {
                     constants.content_box_inset.cross_start(constants.dir) + resolved_margin.cross_start(constants.dir)
                 }
-                (AlignItemsKeyword::Baseline | AlignItemsKeyword::Stretch | AlignItemsKeyword::FlexStart, false)
+                (AlignItemsKeyword::Baseline | AlignItemsKeyword::LastBaseline | AlignItemsKeyword::Stretch | AlignItemsKeyword::FlexStart, false)
                 | (AlignItemsKeyword::FlexEnd, true) => {
                     constants.content_box_inset.cross_start(constants.dir) + resolved_margin.cross_start(constants.dir)
                 }
-                (AlignItemsKeyword::Baseline | AlignItemsKeyword::Stretch | AlignItemsKeyword::FlexStart, true)
+                (AlignItemsKeyword::Baseline | AlignItemsKeyword::LastBaseline | AlignItemsKeyword::Stretch | AlignItemsKeyword::FlexStart, true)
                 | (AlignItemsKeyword::FlexEnd, false) => {
                     constants.container_size.cross(constants.dir)
                         - constants.content_box_inset.cross_end(constants.dir)
diff --git a/src/compute/grid/alignment.rs b/src/compute/grid/alignment.rs
index 8016ca60..2da349bb 100644
--- a/src/compute/grid/alignment.rs
+++ b/src/compute/grid/alignment.rs
@@ -328,6 +328,7 @@ pub(super) fn align_item_within_area(
         AlignItemsKeyword::Start
         | AlignItemsKeyword::FlexStart
         | AlignItemsKeyword::Baseline
+        | AlignItemsKeyword::LastBaseline
         | AlignItemsKeyword::Stretch => {
             if direction.is_rtl() {
                 grid_area_size - resolved_size - resolved_margin.end
diff --git a/src/compute/grid/mod.rs b/src/compute/grid/mod.rs
index 1a631a8f..8ac1b697 100644
--- a/src/compute/grid/mod.rs
+++ b/src/compute/grid/mod.rs
@@ -738,11 +738,31 @@ pub fn compute_grid_layout<Tree: LayoutGridContainer>(
         item.y_position + item.baseline.unwrap_or(item.height)
     };
 
-    LayoutOutput::from_sizes_and_baselines(
+    // Determine the grid container's last baseline from the last row containing items.
+    let grid_container_last_baseline: f32 = {
+        let last_row = items[items.len() - 1].row_indexes.start;
+        let last_row_items = items[..].split(|item| item.row_indexes.start != last_row).last().unwrap();
+        let row_has_baseline_item = last_row_items
+            .iter()
+            .any(|item| item.align_self == AlignSelf::LAST_BASELINE || item.align_self == AlignSelf::BASELINE);
+        let item = if row_has_baseline_item {
+            last_row_items
+                .iter()
+                .find(|item| item.align_self == AlignSelf::LAST_BASELINE || item.align_self == AlignSelf::BASELINE)
+                .unwrap()
+        } else {
+            &last_row_items[0]
+        };
+        item.y_position + item.baseline.unwrap_or(item.height)
+    };
+
+    let mut output = LayoutOutput::from_sizes_and_baselines(
         container_border_box,
         item_content_size_contribution,
         Point { x: None, y: Some(grid_container_baseline) },
-    )
+    );
+    output.last_baselines = Point { x: None, y: Some(grid_container_last_baseline) };
+    output
 }
 
 /// Reverses only non-gutter column tracks in-place while preserving line/gutter slots.
diff --git a/src/compute/leaf.rs b/src/compute/leaf.rs
index 4f1d979c..12ace075 100644
--- a/src/compute/leaf.rs
+++ b/src/compute/leaf.rs
@@ -100,6 +100,7 @@ where
                 #[cfg(feature = "content_size")]
                 content_size: Size::ZERO,
                 first_baselines: Point::NONE,
+                last_baselines: Point::NONE,
                 top_margin: CollapsibleMarginSet::ZERO,
                 bottom_margin: CollapsibleMarginSet::ZERO,
                 margins_can_collapse_through: false,
@@ -155,6 +156,7 @@ where
         #[cfg(feature = "content_size")]
         content_size: measured_size + padding.sum_axes(),
         first_baselines: Point::NONE,
+        last_baselines: Point::NONE,
         top_margin: CollapsibleMarginSet::ZERO,
         bottom_margin: CollapsibleMarginSet::ZERO,
         margins_can_collapse_through: !has_styles_preventing_being_collapsed_through
diff --git a/src/style/alignment.rs b/src/style/alignment.rs
index eda57a0b..6484585a 100644
--- a/src/style/alignment.rs
+++ b/src/style/alignment.rs
@@ -36,8 +36,10 @@ pub enum AlignItemsKeyword {
     FlexEnd,
     /// Items are packed along the center of the cross axis.
     Center,
-    /// Items are aligned such as their baselines align.
+    /// Items are aligned such as their first baselines align.
     Baseline,
+    /// Items are aligned such as their last baselines align.
+    LastBaseline,
     /// Stretch to fill the container.
     Stretch,
 }
@@ -141,8 +143,10 @@ impl AlignItems {
     pub const FLEX_END: Self = Self { keyword: AlignItemsKeyword::FlexEnd, safety: AlignmentSafety::Unsafe };
     /// Items are packed along the center of the cross axis.
     pub const CENTER: Self = Self { keyword: AlignItemsKeyword::Center, safety: AlignmentSafety::Unsafe };
-    /// Items are aligned such as their baselines align.
+    /// Items are aligned such as their first baselines align.
     pub const BASELINE: Self = Self { keyword: AlignItemsKeyword::Baseline, safety: AlignmentSafety::Unsafe };
+    /// Items are aligned such as their last baselines align.
+    pub const LAST_BASELINE: Self = Self { keyword: AlignItemsKeyword::LastBaseline, safety: AlignmentSafety::Unsafe };
     /// Stretch to fill the container.
     pub const STRETCH: Self = Self { keyword: AlignItemsKeyword::Stretch, safety: AlignmentSafety::Unsafe };
     /// Like [`AlignItems::START`], but falls back to [`AlignItems::START`] when the
@@ -207,6 +211,7 @@ impl FromCss for AlignItems {
             "flex-end" => Ok(Self::FLEX_END),
             "center" => Ok(Self::CENTER),
             "baseline" => Ok(Self::BASELINE),
+            "last-baseline" | "last baseline" => Ok(Self::LAST_BASELINE),
             "stretch" => Ok(Self::STRETCH),
             _ => Err(input.new_unexpected_token_error(Token::Ident(first))),
         }
@@ -368,6 +373,7 @@ const ALIGN_ITEMS_NAMES: &[&str] = &[
     "FlexEnd",
     "Center",
     "Baseline",
+    "LastBaseline",
     "Stretch",
     "SafeStart",
     "SafeEnd",
@@ -386,6 +392,7 @@ impl serde::Serialize for AlignItems {
             (AlignItemsKeyword::FlexEnd, AlignmentSafety::Unsafe) => "FlexEnd",
             (AlignItemsKeyword::Center, AlignmentSafety::Unsafe) => "Center",
             (AlignItemsKeyword::Baseline, _) => "Baseline",
+            (AlignItemsKeyword::LastBaseline, _) => "LastBaseline",
             (AlignItemsKeyword::Stretch, _) => "Stretch",
             (AlignItemsKeyword::Start, AlignmentSafety::Safe) => "SafeStart",
             (AlignItemsKeyword::End, AlignmentSafety::Safe) => "SafeEnd",
@@ -414,6 +421,7 @@ impl<'de> serde::Deserialize<'de> for AlignItems {
                     "FlexEnd" => AlignItems::FLEX_END,
                     "Center" => AlignItems::CENTER,
                     "Baseline" => AlignItems::BASELINE,
+                    "LastBaseline" => AlignItems::LAST_BASELINE,
                     "Stretch" => AlignItems::STRETCH,
                     "SafeStart" => AlignItems::SAFE_START,
                     "SafeEnd" => AlignItems::SAFE_END,
diff --git a/src/tree/layout.rs b/src/tree/layout.rs
index 30495370..0405ff67 100644
--- a/src/tree/layout.rs
+++ b/src/tree/layout.rs
@@ -166,6 +166,8 @@ pub struct LayoutOutput {
     pub content_size: Size<f32>,
     /// The first baseline of the node in each dimension, if any
     pub first_baselines: Point<Option<f32>>,
+    /// The last baseline of the node in each dimension, if any
+    pub last_baselines: Point<Option<f32>>,
     /// Top margin that can be collapsed with. This is used for CSS block layout and can be set to
     /// `CollapsibleMarginSet::ZERO` for other layout modes that don't support margin collapsing
     pub top_margin: CollapsibleMarginSet,
@@ -184,6 +186,7 @@ impl LayoutOutput {
         #[cfg(feature = "content_size")]
         content_size: Size::ZERO,
         first_baselines: Point::NONE,
+        last_baselines: Point::NONE,
         top_margin: CollapsibleMarginSet::ZERO,
         bottom_margin: CollapsibleMarginSet::ZERO,
         margins_can_collapse_through: false,
@@ -203,6 +206,7 @@ impl LayoutOutput {
             #[cfg(feature = "content_size")]
             content_size,
             first_baselines,
+            last_baselines: Point::NONE,
             top_margin: CollapsibleMarginSet::ZERO,
             bottom_margin: CollapsibleMarginSet::ZERO,
             margins_can_collapse_through: false,
