diff --git a/tx.go b/tx.go
index 5a4d6f0..821b87d 100644
--- a/tx.go
+++ b/tx.go
@@ -484,11 +484,25 @@ func sameFile(f1, f2 *os.File) (bool, error) {
 // A reader transaction is maintained during the copy so it is safe to continue
 // using the database while a copy is in progress.
 func (tx *Tx) CopyFile(path string, mode os.FileMode) error {
-	f, err := tx.db.openFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, mode)
+	f, err := tx.db.openFile(path, os.O_RDWR|os.O_CREATE, mode)
 	if err != nil {
 		return err
 	}
 
+	same, err := sameFile(tx.db.file, f)
+	if err != nil {
+		_ = f.Close()
+		return err
+	}
+	if same {
+		_ = f.Close()
+		return fmt.Errorf("copy destination is the source database")
+	}
+	if err := f.Truncate(0); err != nil {
+		_ = f.Close()
+		return err
+	}
+
 	_, err = tx.WriteTo(f)
 	if err != nil {
 		_ = f.Close()
